toxcore/network.c

```diff
@@ -1725,7 +1725,17 @@ char *net_new_strerror(int error)
 #else
     char *str = (char *)malloc(256);
 #ifdef _GNU_SOURCE
-    str = strerror_r(error, str, 256);
+    char *retstr = strerror_r(error, str, 256);
+
+    if (retstr != str) {
+        const size_t retstr_len = strlen(retstr) + 1;
+        const size_t copy_len = retstr_len > 256 ? 256 : retstr_len;
```

Comment:
min_u32 maybe?

Reply:
Aha, so we do have a min macro. Actually, it's not even a macro but a function. Feels a bit weird treating `size_t` as `uint32_t`, but I guess it works...